pkg/services/serviceaccounts/api/api.go

```diff
@@ -81,24 +77,23 @@ func (api *ServiceAccountsAPI) RegisterAPIEndpoints(
 
 // POST /api/serviceaccounts
 func (api *ServiceAccountsAPI) CreateServiceAccount(c *models.ReqContext) response.Response {
-	cmd := serviceaccounts.CreateServiceAccountForm{}
+	type createServiceAccountForm struct {
+		Name string `json:"name" binding:"Required"`
+	}
+	cmd := createServiceAccountForm{}
 	if err := web.Bind(c.Req, &cmd); err != nil {
 		return response.Error(http.StatusBadRequest, "Bad request data", err)
 	}
-	cmd.OrgID = c.OrgId
 
-	user, err := api.service.CreateServiceAccount(c.Req.Context(), &cmd)
+	serviceAccount, err := api.store.CreateServiceAccount(c.Req.Context(), c.OrgId, cmd.Name)
 	switch {
-	case errors.Is(err, serviceaccounts.ErrServiceAccountNotFound):
-		return response.Error(http.StatusBadRequest, "Failed to create role with the provided name", err)
+	case errors.Is(err, &database.ErrSAInvalidName{}):
+		return response.Error(http.StatusBadRequest, "Service account name with provided name already exists", err)
```

Comment:
nit: I think the wording here is slightly odd
```suggestion
		return response.Error(http.StatusBadRequest, "Service account with the provided name already exists", err)
```

Reply:
Actually, this is also repeating information that the error itself contains. So maybe It could be shorten to something like `Invalid service account name`?